src/arch/LowLevelWindow/LowLevelWindow_SDL.cpp

```diff
@@ -0,0 +1,258 @@
+
+    // If we're just resizing we should re-centre the window
+    if (!bNewDeviceOut)
+    {
+        SDL_SetWindowPosition(m_pWindow, SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED);
+    }
+
+    if (p.sIconFile)
+    {
+        SetWindowIcon(p.sIconFile);
+    }
+
+    // Set the "Real Width/Height" to the HiDpi values so the scaling is correct
+    m_CurrentVideoParams = ActualVideoModeParams(p, widthInPixels, heightInPixels, false);
+    // Set the refresh rate to the current display's refresh rate
+    // Note: The casting to int can cause refresh rates like 59.<whatever> to be truncated to 59.0.
+    // I'm unsure if we want to just ceil the values here.
+    m_CurrentVideoParams.rate = static_cast<int>(currentDisplayMode->refresh_rate);
+
+    // Success!
+    return "";
+}
+
+/**
+ * Enumerate a list of displays and display resolutions (or modes).
+ * Each display is named, and uses the id: SDL_<order number>.
+ * @param out
+ */
+void LowLevelWindow_SDL::GetDisplaySpecs(DisplaySpecs& out) const
+{
+    int displayCount = 0;
+    const auto displayId = SDL_GetDisplays(&displayCount);
+
+    if (displayCount <= 0)
+    {
+        LOG->Info("[LLW_SDL::GetDisplaySpecs] Failed to query displays!");
+        return;
+    }
+
+    int currentWidth = 0;
+    int currentHeight = 0;
+    SDL_GetWindowSizeInPixels(m_pWindow, &currentWidth, &currentHeight);
+    LOG->Info("[LLW_SDL::GetDisplaySpecs] Current size: [%dx%d]", currentWidth, currentHeight);
+
+    int displayModeCount = 0;
+    for (int di = 0; di < displayCount; di++)
+    {
+        SDL_Rect displayBounds;
+        std::set<DisplayMode> screenModes;
+        const auto displayModes = SDL_GetFullscreenDisplayModes(displayId[di], &displayModeCount);
+
+        SDL_GetDisplayUsableBounds(displayId[di], &displayBounds);
+
+        bool foundCurrentMode = false;
+        DisplayMode currentMode = {};
+        for (int dmi = 0; dmi < displayModeCount; dmi++)
+        {
+            auto mode = displayModes[dmi];
+            DisplayMode m = {
+                static_cast<unsigned int>(mode->w), static_cast<unsigned int>(mode->h), static_cast<double>(mode->refresh_rate)
+            };
+            screenModes.insert(m);
+
+            if (mode->w == currentWidth && mode->h == currentHeight)
+            {
+                currentMode = m;
+                foundCurrentMode = true;
+            }
+        }
+
+        // If we haven't found our existing mode, let's create a dummy entry so we don't assert in debug
+        if (!foundCurrentMode)
+        {
+            // For current refresh rate
+            auto currentDisplayMode = SDL_GetCurrentDisplayMode(displayId[di]);
+            DisplayMode m = {
+                static_cast<unsigned int>(currentWidth), static_cast<unsigned int>(currentHeight),
+                static_cast<double>(currentDisplayMode->refresh_rate)
+            };
+            screenModes.insert(m);
+            currentMode = m;
+        }
```

Comment:
There was some off-by-one scaling errors (i.e. 1280x769). I haven't seen it in a while though, so this might not be needed.